include/fiction/algorithms/simulation/sidb/random_layout_generator.hpp

```diff
@@ -0,0 +1,178 @@
+//
+// Created by Jan Drewniok on 05.04.23.
+//
+
+#ifndef FICTION_RANDOM_LAYOUT_GENERATOR_HPP
+#define FICTION_RANDOM_LAYOUT_GENERATOR_HPP
+
+#include "fiction/algorithms/path_finding/distance.hpp"
+#include "fiction/io/write_sqd_layout.hpp"
+#include "fiction/layouts/cell_level_layout.hpp"
+#include "fiction/technology/cell_technologies.hpp"
+#include "fiction/technology/charge_distribution_surface.hpp"
+#include "fiction/technology/sidb_nm_position.hpp"
+#include "fiction/traits.hpp"
+#include "fiction/types.hpp"
+#include "fiction/utils/hash.hpp"
+#include "fiction/utils/layout_utils.hpp"
+
+#include <cstdint>
+#include <iostream>
+#include <random>
+#include <string_view>
+#include <unordered_set>
+#include <vector>
+
+namespace fiction
+{
+/**
+ * This struct stores the parameters for the *generate_random_layout* algorithm.
+ */
+template <typename Lyt>
+struct random_layout_params
+{
+    /**
+     * Two coordinates that span the region where SiDBs may be placed (order is not important).
+     */
+    std::pair<typename Lyt::cell, typename Lyt::cell> coordinate_pair;
+    /**
+     * Number of SiDBs that are placed on the layout.
+     */
+    uint64_t number_of_sidbs = 0;
+    /**
+     * If positively charged SiDBs should be prevented, SiDBs are not placed closer than the minimal_spacing.
+     */
+    bool prevent_positive_charges = true;
+    /**
+     * If positively charged SiDBs should be prevented, SiDBs are not placed closer than this value (2 cells as
+     * Euclidean distance by default).
+     */
+    double minimal_spacing = 2;
+    /**
+     * Maximal number of steps to place the given number of SiDBs.
+     */
+    uint64_t maximal_attempts = 10E6;
+};
+
+/**
+ * Generates a random layout of SiDBs based on the provided parameters.
+ *
+ * @tparam Lyt The layout type.
+ * @param params The parameters for generating the random layout.
+ * @param lyt_skeleton A layout to which random cells are added (useful if you need to add random cells to a given
+ * layout).
```

Comment:
Since you pass this skeleton, why do you not add the random cells in-place? Why create a new layout?

Reply:
True! Not necessary. Thanks!

Reply:
If necessary, a skeleton can be used as input to which random cells are added. I copy it because I don't want to check the skeleton.